test/integration/distributed-tracing/traceparent-sampled-flag.test.js

```diff
@@ -0,0 +1,390 @@
+// able to force the correct traceparent/tracestate headers.
+const defaultAgentConfig = {
+  instrumentation: {
+    undici: { enabled: false }
+  }
+}
+
+test.beforeEach(ctx => {
+  ctx.nr = {}
+})
+
+test.afterEach(ctx => {
+  ctx.nr.server.close()
+  if (ctx.nr.agent) {
+    helper.unloadAgent(ctx.nr.agent)
+  }
+})
+
+test('remote_parent_sampled: always_on', async t => {
+  const agentConfig = Object.assign({}, defaultAgentConfig, {
+    distributed_tracing: {
+      sampler: {
+        remote_parent_sampled: 'always_on'
+      }
+    }
+  })
+  await beforeEach(t.nr, agentConfig, validator)
+
+  const { host, port } = t.nr
+
+  const result = await doRequest({ host, port })
+  assert.equal(result, 'ok')
+
+  function validator(tx) {
+    assert.equal(tx.acceptedDistributedTrace, true)
+    assert.equal(tx.traceId, TRACE_ID)
+    assert.equal(tx.parentSpanId, SPAN_ID)
+    assert.equal(tx.sampled, true)
+    assert.equal(tx.priority, 2.0)
+  }
+})
+
+test('remote_parent_sampled: always_off', async t => {
+  const agentConfig = Object.assign({}, defaultAgentConfig, {
+    distributed_tracing: {
+      sampler: {
+        remote_parent_sampled: 'always_off'
+      }
+    }
+  })
+  await beforeEach(t.nr, agentConfig, validator)
+
+  const { host, port } = t.nr
+
+  const result = await doRequest({ host, port })
+  assert.equal(result, 'ok')
+
+  function validator(tx) {
+    assert.equal(tx.acceptedDistributedTrace, true)
+    assert.equal(tx.traceId, TRACE_ID)
+    assert.equal(tx.parentSpanId, SPAN_ID)
+    assert.equal(tx.sampled, false)
+    assert.equal(tx.priority, 0)
+  }
+})
+
+test('remote_parent_sampled: default', async t => {
+  const agentConfig = Object.assign({}, defaultAgentConfig, {
+    distributed_tracing: {
+      sampler: {
+        remote_parent_sampled: 'default'
+      }
+    }
+  })
+  await beforeEach(t.nr, agentConfig, validator)
+
+  const { host, port } = t.nr
+
+  const result = await doRequest({ host, port })
+  assert.equal(result, 'ok')
+
+  function validator(tx) {
+    assert.equal(tx.acceptedDistributedTrace, true)
+    assert.equal(tx.traceId, TRACE_ID)
+    assert.equal(tx.parentSpanId, SPAN_ID)
+    assert.equal(tx.sampled, true)
+    assert.equal(tx.priority, null)
+  }
+})
+
+test('remote_parent_sampled: default, remote_parent_not_sampled: always_on (flag true)', async t => {
+  const agentConfig = Object.assign({}, defaultAgentConfig, {
+    distributed_tracing: {
+      sampler: {
+        remote_parent_sampled: 'default',
+        remote_parent_not_sampled: 'always_on'
+      }
+    }
+  })
+  await beforeEach(t.nr, agentConfig, validator)
+
+  const { host, port } = t.nr
+
+  const result = await doRequest({ host, port })
+  assert.equal(result, 'ok')
+
+  function validator(tx) {
+    assert.equal(tx.acceptedDistributedTrace, true)
+    assert.equal(tx.traceId, TRACE_ID)
+    assert.equal(tx.parentSpanId, SPAN_ID)
+    assert.equal(tx.sampled, true)
```

Comment:
this will have to change when you change the code. when it is default it should rely on the tracestate to get the sampled value